src/marqo/marqo_url_resolver.py

```diff
@@ -0,0 +1,30 @@
+import time
+
+import requests
+
+
+class MarqoUrlResolver:
+    def __init__(self, api_key=None, expiration_time: int = 60):
```

Comment:
nit: Are we sure that we are willing to only check for updates after a minute? This feels a little long. For example, someone else creates an index, I will have to wait up to a minute after it is available before I can use it. Perhaps a lower value, like 15 seconds or so?